sdk/resource/env.go

```diff
@@ -82,20 +82,20 @@ func constructOTResources(s string) (*Resource, error) {
 		return Empty(), nil
 	}
 	pairs := strings.Split(s, ",")
-	attrs := []attribute.KeyValue{}
+	var attrs []attribute.KeyValue
 	var invalid []string
 	for _, p := range pairs {
-		field := strings.SplitN(p, "=", 2)
-		if len(field) != 2 {
+		f0, f1, found := strings.Cut(p, "=")
```

Comment:
Can we name these better?
E.g. `k, v` and then later we can `key := strings.TrimSpace(k)` and `val, err := url.QueryUnescape(strings.TrimSpace(v))`